src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/Telemetry/OutOfProcessTelemetryReporter.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.AspNetCore.Razor.Telemetry;
+using Microsoft.VisualStudio.Telemetry;
+
+namespace Microsoft.CodeAnalysis.Remote.Razor.Telemetry;
+
+[Export(typeof(ITelemetryReporter)), Shared]
+internal class OutOfProcessTelemetryReporter : TelemetryReporter
+{
+    public OutOfProcessTelemetryReporter()
```

Comment:
>OutOfProcessTelemetryReporter [](http://example.com/codeflow?start=11&length=29)

Nit: should we use primary constructor here?

Reply:
lol, I did that originally but it looked _too_ short. Felt better to let the code breathe a bit. Happy to take a vote though, `internal class OutOfProcessTelemetryReporter() : TelemetryReporter([TelemetryService.DefaultSession]);` totally would work.

Reply:
I'm fine either way. To be quite honest, I'm not a huge fan of primary constructors so far, especially since we are still using separately defined fields with underscore prefix, which seems to defeat some of the purpose.